src/base/searchengine.cpp

```diff
@@ -104,19 +105,25 @@ QStringList SearchEngine::enabledPlugins() const
     return plugins;
 }
 
-QStringList SearchEngine::supportedCategories() const
+QStringList SearchEngine::supportedCategories(const QString &name) const
```

Comment:
Actually, I don't like this logic here since it lead to inconsistency. What does this param mean in terms of SearchEngine class?
Besides, it's all there:
1. Supported categories for "enabled" plugins is current method behavior.
2. Supported categories for "all" plugins is meaningless since categories provided by disabled plugins only are Unsupported.
3. Supported categories for some particular plugin you can obtain from its PluginInfo.

Reply:
In the current implementation the list of possible categories is already derived from the selected plugins, given that each plugin specifies which categories it supports. Therefore the plugins and categories comboboxes work in conjunction to craft a search query. Should the plugins combobox show categories that aren’t relevant to the plugins being used? I would think not.

Reply:
>Should the plugins combobox show categories that aren’t relevant to the plugins being used? I would think not.

I think the same. I just want this logic to be implemented in correct place.

Reply:
> 2. Supported categories for "all" plugins is meaningless since categories provided by disabled plugins only are Unsupported.

When "All plugins" is selected, disabled plugins are used in the search. Their supported categories are relevant.

Reply:
I always "hate" the presence of such inconsistencies in legacy app parts. But even more surprising to me why many developers prefer not to fix these inconsistencies, but only aggravate them.
Why we need "disabled" plugin state, if we can simply omit this state and use all plugins? It requires this feature to be either removed or used in consistent way (I mean ability to enable/disable plugins). Personally I prefer it to be removed and we just have "selected plugins" behavior at UI side.
@Chocobo1, please share this discussion, I need your opinion.

Reply:
>Personally I prefer it to be removed and we just have "selected plugins" behavior at UI side.

What to be removed? "disabled" plugin state? If so, I prefer not touching it to avoid user rant (if there are any) later on.

Reply:
>I prefer not touching it to avoid user rant (if there are any) later on.

I do not insist on doing just that. I just want to get rid of inconsistencies, or at least not to produce new.

Reply:
>I just want to get rid of inconsistencies, or at least not to produce new.

Pardon, the inconsistencies are not obvious to me even after reading this thread, care to point out?

Reply:
SearchEngine allows to **disable** some installed plugins. When we ask for supported search categories it returns only categories from enabled plugins (it's consistent).
But it allows to search using all installed plugins (including disabled). It's first inconsistency. The second confusing thing is in GUI. There is one additional item in plugins combobox to allow start search using several plugins, but really it show plugins dialog (apparently, in order the user enable/disable some plugins).
Maybe you guys think this is correct behavior?

Reply:
> When we ask for supported search categories it returns only categories from enabled plugins (it's consistent).

How is this more "consistent" than only returning categories supported by the plugins actually being used for the search?

> There is one additional item in plugins combobox to allow start search using several plugins, but really it show plugins dialog (apparently, in order the user enable/disable some plugins).

I too don't understand the purpose of this option. The "Search plugins..." button has the same effect and is more explicit. I wouldn't be opposed to removing this option from the combobox.

Reply:
>How is this more "consistent" than only returning categories supported by the plugins actually being used for the search?

Because there is no "the plugins actually being used for the search" out of some particular "search job".
Current State of SearchEngine is defined by the set of currently enabled plugins. Each property should be constant for the same object state (i.e. have the same value each time it accessed).
If you can obtain set of categories supported by some arbitrary set of plugins you can use its descriptions (PluginInfo's) directly.